bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+class FunctionSignature():
+    """
+    Class to determine a Function's Signature.
+    """
+    function_cache = {} # Variable to store previous functions so as to not rejit.
+    def __init__(self, rdf = None) -> None:
+        if rdf != None:
+            self.rdf = rdf
+            self.col_names = rdf.GetColumnNames()
+        else:
+            self.rdf = None
+            self.col_names = []
+    
+    def find_type(self, x, rdf = None):
+        """
+        Function to determine the type of a variable x.
+        If x is a string. It maps it to the corresponding column's type in RDF. (String constants are not supported)
+        Else it determines the fundemental type of x. (int, float, double)
+        Else it is an a numpy array and maps it to corresponding RVec.
+        Otherwise falgs a type error
+        Args:
+	    `   x: Variable whose type is to be determined
+
+        Returns:
+            type of x
+        """
+        try:
+            import numpy as np
+        except:
+            raise ImportError("Failed to import numpy during call to determine function signature.")
+        if isinstance(x, str):
+            #Can be string constant or can be column name
+            if x in self.col_names: # If x is a column
+                t = self.rdf.GetColumnType(x)
+                if t in map_tree_to_numbadeclares.keys(): # The column is a fundemental type from tree  
+                    return map_tree_to_numbadeclares[t]
+                elif '<' in  t: # The column type is a RVec<type>
+                    g = re.match('(.*)<(.*)>', t).groups(0)
+                    if g[1] in map_tree_to_numbadeclares.keys():
+                        return "RVec<" + map_tree_to_numbadeclares[g[1]] + ">"
+                    return "RVec<" + str(g[1]) + ">" # There are data type that leak into here. NOt sure from where. BUt need to implement something here.
+                    
+                else:
+                    return t
+            else :
+                return 'str'
+                #! Numba Declare does not support "string" type. Check _numbadeclare.Thus, Cannot pass string constants to the filter/Defines..
+        elif isinstance(x, bool):
+            return 'bool'
+        elif isinstance(x, int):
+            return 'int'
+        elif isinstance(x, float):
+            return 'float'
+        elif isinstance(x, np.ndarray): 
```

Comment:
Here and below, whenever you have a list of `if elif` statements, maybe try substituting it with a dictionary pattern:
```python
types = {float: "float", bool: "bool", int: "int"}

def find_type(x):
    return types[type(x)]
```

Reply:
Done in find_types.